packages/wrangler/src/__tests__/secret.test.ts

```diff
@@ -690,4 +527,217 @@ describe("wrangler secret", () => {
 			);
 		});
 	});
+
+	describe("secret:bulk", () => {
+		it("should use secret:bulk w/ pipe input", async () => {
+			writeWranglerToml();
+			let counter = 0;
+			msw.use(
+				rest.put(
+					`*/accounts/:accountId/workers/scripts/:scriptName/secrets`,
+					(req, res, ctx) => {
+						expect(req.params.accountId).toEqual("some-account-id");
+						counter++;
+
+						return res(
+							ctx.json(
+								createFetchResult({
+									name: `secret-name-${counter}`,
+									type: "secret_text",
+								})
+							)
+						);
+					}
+				)
+			);
+
+			await runWrangler(`secret:bulk`);
+			expect(std.out).toMatchInlineSnapshot(`
+					"🌀 Creating the secrets for the Worker \\"test-name\\"
+					✨ Successfully created secret for key: secret1
+					✨ Successfully created secret for key: password
```

Comment:
in the msw handler above, the secret names are `secret-name-${counter}` but that doesn't match these? took me a while to realise the mock data is coming from the readline mock at the top of the file

might make this test easier to understand if the mock data was specified here within the test, and returned by the mock at the top of file – but if that gets messy, I wouldn't lose sleep leaving it like this

Reply:
Yeah to be honest this whole testing situation for this was a huge pain.